### PR TITLE
Add queue config flag to Thrift pool

### PR DESCRIPTION
**Pull request by SuperQ**
Add a `fifo_queue` flag to the `thrift_pool_from_config` constructor.
This allows easy runtime configuration of the `queue_cls`.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Review by ConorMcGee — APPROVED** [review]

File: baseplate/lib/thrift_pool.py
Comment (MelissaCole) [inline baseplate/lib/thrift_pool.py]: `:` -> `,`

File: baseplate/lib/thrift_pool.py
Comment (MelissaCole) [inline baseplate/lib/thrift_pool.py:97]: Could you add this to the supported keys list above? Another option could be to accept a string defining what type of queue class to use in case more are added. Something like `"queue_cls": config.Optional(config.String, default="fifo"),` or if there's a way to specify a list of available options but I don't remember if that syntax is available atm

**Review by MelissaCole — CHANGES_REQUESTED** [review]
Looks good - typo and documenting the new arg

File: baseplate/lib/thrift_pool.py
Comment (SuperQ) [reply baseplate/lib/thrift_pool.py]: Whups!

File: baseplate/lib/thrift_pool.py
Comment (SuperQ) [reply baseplate/lib/thrift_pool.py:97]: I think the only options we really need to support here are LIFO and FIFO. We'd have to completely replace the scheduler if we wanted to do something more fancy.

File: tests/unit/lib/thrift_pool_tests.py
Comment (MelissaCole) [inline tests/unit/lib/thrift_pool_tests.py]: `=` -> `:`

Also you might want to represent it as a string, so `"fifo-test.fifo_queue": "true",`

**Review by MelissaCole — CHANGES_REQUESTED** [review]
Looks good - just a typo remaining

File: tests/unit/lib/thrift_pool_tests.py
Comment (SuperQ) [reply tests/unit/lib/thrift_pool_tests.py]: Fixed.

File: tests/unit/lib/thrift_pool_tests.py
Comment (MelissaCole) [inline tests/unit/lib/thrift_pool_tests.py]: nit: linter doesn't like the extra line

**Review by MelissaCole — APPROVED** [review]
lgtm after fixing it up for the linter

File: tests/unit/lib/thrift_pool_tests.py
Comment (SuperQ) [reply tests/unit/lib/thrift_pool_tests.py]: *sigh*

File: tests/unit/lib/thrift_pool_tests.py
Comment (MelissaCole) [inline tests/unit/lib/thrift_pool_tests.py]: A couple more changes:
* remove `import queue` on L1
* change `gevent.queue` to `gevent_queue` on L130 (but would then conflict with [this PR](https://github.com/reddit/baseplate.py/pull/656/files#diff-3de570a9b43ce8e7774ac672c91e3f49f999a4d88310d9afb12bd997a840a6fdL7))

File: tests/unit/lib/thrift_pool_tests.py
Comment (MelissaCole) [inline tests/unit/lib/thrift_pool_tests.py]: This conflicts with https://github.com/reddit/baseplate.py/pull/656/files#diff-3de570a9b43ce8e7774ac672c91e3f49f999a4d88310d9afb12bd997a840a6fdR299. What'd be the best way to resolve it?

cc @RoganMurley

**Review by MelissaCole — CHANGES_REQUESTED** [review]
lgtm but a couple fixes for tests

File: tests/unit/lib/thrift_pool_tests.py
Comment (SuperQ) [reply tests/unit/lib/thrift_pool_tests.py]: I could wait till #656 is merged and rebase this.

File: tests/unit/lib/thrift_pool_tests.py
Comment (SuperQ) [reply tests/unit/lib/thrift_pool_tests.py]: I flipped this around to remove my import change and fixed the test to use `gevent.queue`.

File: tests/unit/lib/thrift_pool_tests.py
Comment (SuperQ) [reply tests/unit/lib/thrift_pool_tests.py]: I reverted my change here.

File: tests/unit/lib/thrift_pool_tests.py
Comment (MelissaCole) [inline tests/unit/lib/thrift_pool_tests.py]: `True` -> `'true'`

Comment (SuperQ) [reply tests/unit/lib/thrift_pool_tests.py]: Fixed.

**Review by MelissaCole — APPROVED** [review]

**Pull request merged**
